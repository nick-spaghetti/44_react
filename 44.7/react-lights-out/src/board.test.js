import React from "react";
import Board from "./Board";
import { render, fireEvent } from "@testing-library/react";

describe("<Board /> rendering", () => {
	it("should render without breaking", () => {
		render(<Board />);
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<Board />);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should show text", () => {
		const { getByText } = render(<Board chanceLightStartsOn={0} />);
		expect(getByText("you win")).toHaveTextContent("you win");
		expect(getByText("you win")).toBeInTheDocument();
	});
});

describe("cell click", () => {
	it("toggles lights", () => {
		const { getAllByRole } = render(
			<Board
				nrows={3}
				ncols={3}
				chanceLightStartsOn={1}
			/>
		);
		const cells = getAllByRole("button");
		cells.forEach((cell) => {
			expect(cell).toHaveClass("Cell-lit");
		});
		fireEvent.click(cells[4]);
		let litIndices = [0, 2, 6, 8];
		cells.forEach((cell, idx) => {
			if (litIndices.includes(idx)) {
				expect(cell).toHaveClass("Cell-lit");
			} else {
				expect(cell).not.toHaveClass("Cell-lit");
			}
		});
	});
	it("says you won", () => {
		const { queryByText, getAllByRole } = render(
			<Board
				nrows={1}
				ncols={3}
				chanceLightStartsOn={1}
			/>
		);
		expect(queryByText("you win")).not.toBeInTheDocument();
		const cells = getAllByRole("button");
		fireEvent.click(cells[1]);
		expect(queryByText("you win")).toBeInTheDocument();
	});
});
