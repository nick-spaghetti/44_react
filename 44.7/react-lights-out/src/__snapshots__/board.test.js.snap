// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> rendering matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      you win
    </h1>
  </div>
</DocumentFragment>
`;
